### PR TITLE
Request for Freertos queue branch

### PR DESCRIPTION
**Pull request by priyankakumari22k**
Request for freeRTOS_Queue task branch

### CODE REVIEW

File: Src/main.c [hunk 1]
```diff
@@ -274,76 +253,69 @@ static void MX_ICACHE_Init(void)
...
-    }
-
-
-  /* USER CODE END LED_Thread2 */
+	char *pcReceivedString;
+	//BaseType_t xStatus;
+	 for( ;; )
+	 {
+
+		 if( uxQueueMessagesWaiting( xPointerQueue ) == 5 )
```
Comment (akashk-eic) [inline Src/main.c]: why same if condition required 2 times?

File: Src/main.c [hunk 2]
```diff
@@ -274,76 +253,69 @@ static void MX_ICACHE_Init(void)
...
+	  /* Increment the counter so the string is different on each iteration of this task. */
+	  xStringNumber++;
+
+	  xStatus= xQueueSend( xPointerQueue,&pcStringToSend, portMAX_DELAY );
+
+	  if( xStatus != pdPASS )
+	   {
+	   /* The send operation could not complete because the queue was full -this must be an error as the queue should never contain more than
+	   one item! */
+	   printf( "Could not send to the queue.\r\n" );
```
Comment (akashk-eic) [inline Src/main.c]: Code indentation (Tab/Space) should be same for all project.

You can use STM IDE auto indent feature.

File: Src/main.c [hunk 3]
```diff
@@ -274,76 +253,69 @@ static void MX_ICACHE_Init(void)
 
 /* USER CODE END 4 */
 
-/* USER CODE BEGIN Header_LED_Thread1 */
-/**
```
Comment (akashk-eic) [inline Src/main.c:278]: keep practice of adding this kind of Doxygen format comment on top of all newly added function's definition.

/*
  * @brief  Function description
  * @param  arguments info
  * @retval return value info
  */

**Review by akashk-eic — CHANGES_REQUESTED** [review]

File: Src/main.c (hunk 1 above)
Comment (priyankakumari22k) [reply Src/main.c]: other if function is for checking queue is empty or not. I added the comment line above each statement.

File: Src/main.c (hunk 2 above)
Comment (priyankakumari22k) [reply Src/main.c]: changes completed

File: Src/main.c (hunk 3 above)
Comment (priyankakumari22k) [reply Src/main.c:278]: comments are added above the function

File: Src/main.c (hunk 1 above)
Comment (akashk-eic) [reply Src/main.c]: I mean to say here, can we avoid API calling 2 times? By storing return value for if condition. API calling multiple times will unnecessarily add overhead execution time.

File: Src/main.c [hunk 4]
```diff
@@ -131,51 +122,39 @@ int main(void)
...
-  /* add events, ... */
-  /* USER CODE END RTOS_EVENTS */
+  /* The queue is created to hold a maximum of 5 values, each of which is
+   large enough to hold a variable of type Data_t ). */
+   //xQueue = xQueueCreate( 3, sizeof( Data_t )) ;
+   xPointerQueue = xQueueCreate( 5, sizeof( char * ) );
+
+   if( xPointerQueue != NULL )
+   {
+   /* Create two instances of the task that will send to the queue. The
```
Comment (akashk-eic) [inline Src/main.c:133]: Still code indentation in not proper.
You can enable trailing whitespace visibility in IDE from editors configuration.

**Review by akashk-eic — CHANGES_REQUESTED** [review]

File: Src/main.c (hunk 4 above)
Comment (priyankakumari22k) [reply Src/main.c:133]: enable the whitespace option

Comment (priyankakumari22k) [reply Src/main.c:133]: Made the changes

File: Src/main.c [hunk 5]
```diff
@@ -274,76 +253,85 @@ static void MX_ICACHE_Init(void)
...
+		 num=uxQueueMessagesWaiting( xPointerQueue );
+         //for checking queue is full or not
+		 if( num == 5 )
+		 	 {
+			 printf( "Queue is full!\r\n" );
+		 	 }
+
+
+		 //for checking queue is empty or not
+		 if( num == 0 )
```
Comment (akashk-eic) [inline Src/main.c]: can we use else if here? 

If yes, what is the benefit of using else if?

**Review by akashk-eic — CHANGES_REQUESTED** [review]

File: Src/main.c (hunk 5 above)
Comment (priyankakumari22k) [reply Src/main.c]: Made the changes

**Review by akashk-eic — APPROVED** [review]
Looks good.

**Pull request merged**